### PR TITLE
cloud: support backup to s3 when object lock enabled. 

### PR DESCRIPTION
**Pull request by 3pointer**
Signed-off-by: 3pointer <luancheng@pingcap.com>

### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/13442

What's Changed:
when detected object enable for bucket from br. tikv will calculate md5 for the content to s3. 

``` support backup to s3 bucket when object lock enabled.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
Test it in EKS cluster.
before this PR we get an error.
```
<Error><Code>InvalidRequest</Code><Message>Content-MD5 OR x-amz-checksum- HTTP header is required for Put Object request
```

after this PR and br updated. backup succeed.

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Support backup to s3 bucket with object lock enabled.
```

### CODE REVIEW

File: components/cloud/aws/src/s3.rs [hunk 1]
```diff
@@ -296,11 +300,21 @@ impl<'client> S3Uploader<'client> {
...
             storage_class: config.storage_class.as_ref().cloned(),
             multi_part_size: config.multi_part_size,
+            object_lock_enabled: config.object_lock_enabled,
             upload_id: "".to_owned(),
             parts: Vec::new(),
         }
     }
 
+    fn get_content_md5(&self, content: &[u8]) -> Option<String> {
+        if self.object_lock_enabled {
```
Comment (YuJuncen) [inline components/cloud/aws/src/s3.rs:306]: Maybe [`bool::then`](http://doc.rust-lang.org/1.63.0/core/primitive.bool.html#method.then)?

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: components/cloud/aws/src/s3.rs (hunk 1 above)
Comment (3pointer) [reply components/cloud/aws/src/s3.rs]: fixed

**Review by overvenus — COMMENTED** [review]
Please add a unit test.

**Comment by 3pointer** [comment]
> Please add a unit test.

done, PTAL

**Review by overvenus — APPROVED** [review]

**Pull request merged**